plugins/USBPrinting/USBPrinterOutputDeviceManager.py

```diff
@@ -128,8 +128,12 @@ def getInstance(cls, engine = None, script_engine = None):
         return USBPrinterOutputDeviceManager._instance
 
     def _getDefaultFirmwareName(self):
-        machine_instance = Application.getInstance().getMachineManager().getActiveMachineInstance()
-        machine_type = machine_instance.getMachineDefinition().getId()
+        # Detecting id of the current machine
+        machine_id = Application.getInstance().getGlobalContainerStack().getBottom().id
```

Comment:
Note that getGlobalContainerStack() can return None. I am not sure if this method can be called in any situation where that is the case but if so, you might want to check for it.